Entwicklung/Cindy/HUSST_Versorgungsdaten_Cindy.xsd

```diff
@@ -1775,35 +1769,35 @@ Sie muss aber mit der ID_Tarifgebiet der Preisstufe übereinstimmen.</xs:documen
 					</xs:documentation>
 				</xs:annotation>
 			</xs:element>
-			<xs:element name="DynAttribut" type="husst:DynAttribut_Subtype" nillable="false" maxOccurs="unbounded" minOccurs="0"/>
+			<xs:element name="DynAttribut" type="husstDV:DynAttribut_Subtype" nillable="false" maxOccurs="unbounded" minOccurs="0"/>
 		</xs:sequence>
 	</xs:complexType>
 
 	<xs:complexType name="Preisstufen_Type">
 		<xs:annotation>
 			<xs:appinfo>
-				<primekey>
-					<field name="ID_Preisstufe"/>
-					<field name="ID_Zeitraum"/>
-				</primekey>
-				<index name="Idx_Preisstufennummer">
-					<field name="ID_Preisstufennummer"/>
-					<field name="ID_Tarifgebiet"/>
-					<field name="ID_Zeitraum"/>
-				</index>
-				<schema name="Tarif" nillable="false"/>
+				<api:primekey>
+					<api:field name="ID_Preisstufe"/>
+					<api:field name="ID_Zeitraum"/>
+				</api:primekey>
+				<api:uniquekey name="Uix_Preisstufen">
+					<api:field name="Preisstufennummer"/>
```

Comment:
Ich glaube der Unique Key funtioniert nicht, da Preisstufennummer nicht gefüllt sein muss (minOccurs="0")

Reply:
für SQLite funktioniert es tatsächlich, SQLite kann auch mit NULL Feldern in Unique Keys umgehen.

Reply:
Lassen wir das dann so?

Reply:
Was passiert denn, wenn in der Sqlite für zwei Preisstufen in einem Tarifgebiet/Zeitraum die Nummer null ist.

Un wir geben ja nicht als Datenbank Sqlite vor. In anderen Datenbanksystemen könnte es daher Probleme geben.
Was meint denn @highQ-Gumbert ?

Reply:
Wenn ich das richtig recherchiert habe, kann der MS SQL Server ab Version 2008+ auch damit umgehen. Können wir nicht einfach einen eigenen HUSST-uniquekey (bspw. :"uniquekeynull") definieren, der NULL-Werte erlaubt? Dann könnten Nutzer, die die Keys verwenden wollen, selbst anhand der bei ihnen eingesetzten Datenbank entscheiden ob zusätzlich zu herkömmlichen uniquekeys auch "uniquekeynull" genutzt werden...

Reply:
Ich kann das in die Beschreibung des uniquekey Elements gerne mit aufnehmen:
	<xs:complexType name="uniquekey_Type">
		<xs:annotation>
			<xs:documentation>Eindeutigkeitsschlüssel Definition. 
			  Liste von einem oder mehreren Datenfelder, die einen Eindeutigkeitsschlüssel der Datenmenge bilden.
			  Neben dem häufig technisch definierten Primärschlüssel, kann es ein oder mehrere datenlogische Schlüssel geben,
			  über denen eine Datenmenge eindeutig sein muss.
			  Für einen Datentyp dürfen mehrere Eindeutigkeitsschlüssel definiert sein.
			  Ein Eindeutigkeitsschlüssel darf auch optionale Felder enthalten. Der fehlende Wert ist dann *eine* mögliche 
			  Ausprägung des Wertes.  
			</xs:documentation>
		</xs:annotation>

Reply:
Ich habe die Änderung noch aufgenommen in die HUSST_Appinfo_Cindy.xsd

Reply:
Das heißt ich darf in einem Zeitraum nur einen Wert ohne Preisstufennummer, also Preisstufennummer = null haben?

Reply:
äh-ja. Das ist wohl eher doof. Ich lösche den Uix. Danke.